test/e2e_node/builder/build.go

```diff
@@ -42,13 +45,34 @@ var buildTargets = []string{
 
 // BuildGo builds k8s binaries.
 func BuildGo() error {
+	if err := BuildTargets(true); err != nil {
+		return fmt.Errorf("unable to build cgo targets : %w", err)
+	}
+	if err := BuildTargets(false); err != nil {
+		return fmt.Errorf("unable to build non-cgo targets : %w", err)
+	}
+	return nil
+}
+
+// BuildGo builds k8s binaries.
+func BuildTargets(cgo bool) error {
 	klog.Infof("Building k8s binaries...")
 	k8sRoot, err := utils.GetK8sRootDir()
 	if err != nil {
 		return fmt.Errorf("failed to locate kubernetes root directory %v", err)
 	}
-	targets := strings.Join(buildTargets, " ")
-	cmd := exec.Command("make", "-C", k8sRoot, fmt.Sprintf("WHAT=%s", targets))
+	targets := buildCGOTargets
+	if !cgo {
+		targets = buildNoCGOTargets
+	}
+	what := strings.Join(targets, " ")
+	cmd := exec.Command("make", "-C", k8sRoot,
+		fmt.Sprintf("WHAT=%s", what))
+	if cgo {
```

Comment:
hmm you shouldn't need to set this? we should be defaulting this per-binary already

Reply:
See `hack/lib/golang.sh`

A) kubelet is a default non-static binary
B) if we are going to override for some reason, `KUBE_CGO_OVERRIDES_LIST` seems more appropriate, but back to A) we should already be doing that.

Is there a bug somewhere causing CGO_ENABLED to get clobbered earlier on or something?

Reply:
@BenTheElder this flows through another code path and does not pick up the list. i tested this explicitly (snippet below) to make sure this code path works with the explicit `CGO_ENABLED` on the command line.

If we need to do something differently, we can consider that outside of this PR as a follow up. i was going with minimum changes as this stuff is fragile needless to say.

```
[1040483:1040482 - 0:1721] 11:15:10 [davanum@ip-172-31-47-86:/dev/pts/0 +1] ~/go/src/k8s.io/kubernetes
$ make KUBE_VERBOSE=100 WHAT=github.com/onsi/ginkgo/v2/ginkgo CGO_ENABLED=1
go version go1.21.1 linux/amd64
Go version: go version go1.21.1 linux/amd64
+++ [0926 11:15:35] Building go targets for linux/amd64
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
Env for linux/amd64: GOOS=linux GOARCH=amd64 GOROOT=/home/davanum/go/src/k8s.io/kubernetes/_output/local/.gimme/versions/go1.21.1.linux.amd64 CGO_ENABLED=1 CC=
Building binaries with GCFLAGS=all=-trimpath=/home/davanum/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes  ASMFLAGS=all=-trimpath=/home/davanum/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes LDFLAGS=all=-X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.buildDate=2023-09-26T11:15:35Z' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.buildDate=2023-09-26T11:15:35Z' -X 'k8s.io/client-go/pkg/version.buildDate=2023-09-26T11:15:35Z' -X 'k8s.io/component-base/version.buildDate=2023-09-26T11:15:35Z' -X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitCommit=46835f8792dfb4a17345e592d1325bf63bc054e4' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.gitCommit=46835f8792dfb4a17345e592d1325bf63bc054e4' -X 'k8s.io/client-go/pkg/version.gitCommit=46835f8792dfb4a17345e592d1325bf63bc054e4' -X 'k8s.io/component-base/version.gitCommit=46835f8792dfb4a17345e592d1325bf63bc054e4' -X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitTreeState=clean' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.gitTreeState=clean' -X 'k8s.io/client-go/pkg/version.gitTreeState=clean' -X 'k8s.io/component-base/version.gitTreeState=clean' -X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitVersion=v1.29.0-alpha.1.5+46835f8792dfb4' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.gitVersion=v1.29.0-alpha.1.5+46835f8792dfb4' -X 'k8s.io/client-go/pkg/version.gitVersion=v1.29.0-alpha.1.5+46835f8792dfb4' -X 'k8s.io/component-base/version.gitVersion=v1.29.0-alpha.1.5+46835f8792dfb4' -X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitMajor=1' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.gitMajor=1' -X 'k8s.io/client-go/pkg/version.gitMajor=1' -X 'k8s.io/component-base/version.gitMajor=1' -X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitMinor=29+' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.gitMinor=29+' -X 'k8s.io/client-go/pkg/version.gitMinor=29+' -X 'k8s.io/component-base/version.gitMinor=29+'  -s -w
Coverage is disabled.
+++ [0926 11:15:41] Placing binaries
1
[1040483:1040482 - 0:1722] 11:15:41 [davanum@ip-172-31-47-86:/dev/pts/0 +1] ~/go/src/k8s.io/kubernetes
$ make KUBE_VERBOSE=100 WHAT=github.com/onsi/ginkgo/v2/ginkgo CGO_ENABLED=0
go version go1.21.1 linux/amd64
Go version: go version go1.21.1 linux/amd64
+++ [0926 11:15:43] Building go targets for linux/amd64
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
Env for linux/amd64: GOOS=linux GOARCH=amd64 GOROOT=/home/davanum/go/src/k8s.io/kubernetes/_output/local/.gimme/versions/go1.21.1.linux.amd64 CGO_ENABLED=0 CC=
Building binaries with GCFLAGS=all=-trimpath=/home/davanum/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes  ASMFLAGS=all=-trimpath=/home/davanum/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes LDFLAGS=all=-X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.buildDate=2023-09-26T11:15:43Z' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.buildDate=2023-09-26T11:15:43Z' -X 'k8s.io/client-go/pkg/version.buildDate=2023-09-26T11:15:43Z' -X 'k8s.io/component-base/version.buildDate=2023-09-26T11:15:43Z' -X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitCommit=46835f8792dfb4a17345e592d1325bf63bc054e4' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.gitCommit=46835f8792dfb4a17345e592d1325bf63bc054e4' -X 'k8s.io/client-go/pkg/version.gitCommit=46835f8792dfb4a17345e592d1325bf63bc054e4' -X 'k8s.io/component-base/version.gitCommit=46835f8792dfb4a17345e592d1325bf63bc054e4' -X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitTreeState=clean' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.gitTreeState=clean' -X 'k8s.io/client-go/pkg/version.gitTreeState=clean' -X 'k8s.io/component-base/version.gitTreeState=clean' -X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitVersion=v1.29.0-alpha.1.5+46835f8792dfb4' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.gitVersion=v1.29.0-alpha.1.5+46835f8792dfb4' -X 'k8s.io/client-go/pkg/version.gitVersion=v1.29.0-alpha.1.5+46835f8792dfb4' -X 'k8s.io/component-base/version.gitVersion=v1.29.0-alpha.1.5+46835f8792dfb4' -X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitMajor=1' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.gitMajor=1' -X 'k8s.io/client-go/pkg/version.gitMajor=1' -X 'k8s.io/component-base/version.gitMajor=1' -X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitMinor=29+' -X 'k8s.io/kubernetes/vendor/k8s.io/component-base/version.gitMinor=29+' -X 'k8s.io/client-go/pkg/version.gitMinor=29+' -X 'k8s.io/component-base/version.gitMinor=29+'  -s -w
Coverage is disabled.
+++ [0926 11:15:46] Placing binaries
```